glib/Timeout.cs

```diff
@@ -28,69 +28,29 @@ namespace GLib {
 
 	public class Timeout {
 
-		[UnmanagedFunctionPointer (CallingConvention.Cdecl)]
-		delegate bool TimeoutHandlerInternal (IntPtr ptr);
-
 		internal class TimeoutProxy : SourceProxy {
+			readonly TimeoutHandler real_handler;
 			public TimeoutProxy (TimeoutHandler real)
 			{
 				real_handler = real;
-				proxy_handler = new TimeoutHandlerInternal (Handler);
-			}
-
-			~TimeoutProxy ()
-			{
-				Dispose (false);
-			}
-
-			public void Dispose ()
-			{
-				Dispose (true);
-				GC.SuppressFinalize (this);
-			}
-
-			protected virtual void Dispose (bool disposing)
```

Comment:
This is an API breakage for sub-classes of the Timeout class, not sure if anyone is doing that